### ISSUE TITLE
SHM-2147: Automation that needs more planning

### ISSUE THREAD

**Issue by chrischabot** [shardeum-mirror/shardeum#130]
field | value
| - | - |
url | https://linear.app/shm/issue/SHM-2147/automation-that-needs-more-planning
author | @
date | Mon Aug  7 04:20:31 BST 2023
state | Canceled
project | 
priority | Low
assignee | @
labels | 
related | 
parent | 
children | 
PRs | 
attachments | 

TODO needs more planning

4 **Need a way to figure out which fully OOS account is the oldest.**

```
This can be on the scan page but will involve more that just the data set available.

I think it will read the “report.txt” json from the oos report.  It will look at each account listed and then ask a node for the timestamp to find the oldest.  If there are multiple different hashes it needs to ask one node from each represented hash.   We may later ask for a different version of this tool that uses downloaded .sqlite dbs of all nodes, but for this request it should use the /account-report endpoint (or even better a new debug endpoint that gives you timestamps for a list of accounts.. )
```

This new debug endpoint may be something that is good for someone else to work on in parallel. TODO make a separate task.

5 **“Automation” support to run data pulling and queries of all the data.**

I think we need a bit of “glue” to make this happen, but the semi-automatic process now is this:

I pick a dev folder, usually named dev_radix_0 (could also be dev or prod2, and the “0” could be any radix of any length.

Then I run the manual command /debug-patcher-dumpTree-partial?radix=000&subtree=true to go with the correct radix set. in this case would be “0” instead of “000”

Then i go to the scan tab and type in the folder “dev_radix_0’ and let the scan run.

This is super helpful, in some cases we want to get everything, so the above could be automated. Note this may take a long time on networks with a lot of data, but that is ok. It will still be helpful as a single click operation that gives us tons of data insights.

Other notes: /debug-patcher-fail-hashes is a good prepass that can be called to one node and tell you which scan level hashes are actually OOS. The process above can skip non OOS scan level hashes.

Note two. eventually scan level hashes will have to be 2 or more characters long. this increase the loops.

Note three: eventually in a sharded network only some nodes will know some of the hashses. The good thing is that we can mostly get this list of nodes from the /debug-patcher-fail-hashesm report. There may be some nodes that are having issues and do not submit a report.. I dont ahve a good answer for this yet.

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2147/automation-that-needs-more-planning#comment-4038a003
author | @
date | Mon Aug  7 04:20:47 BST 2023

![image.png](https://uploads.linear.app/1065affa-aebb-4031-b7e5-3fb32c4187d8/a54f71e1-0c2f-4b78-9281-0c6479335221/44c96c63-8c8c-4534-ba78-c668813f0871)
